chia/cmds/show.py

```diff
@@ -1,13 +1,45 @@
-from typing import Any, Optional, Union, Dict
+from datetime import datetime
+from decimal import Decimal
+from typing import Any, Callable, Dict, Optional, Union
 
-from chia.types.blockchain_format.sized_bytes import bytes32
 import click
 
+from chia.cmds.units import units
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.coin_record import CoinRecord
+from chia.util.bech32m import encode_puzzle_hash
+from chia.util.ints import uint64
 from chia.util.network import is_trusted_inner
 
 
-async def print_connections(client, time, NodeType, trusted_peers: Dict):
-    connections = await client.get_connections()
+def print_coins(
+    verbose: bool,
+    name: str,
+    address_prefix: str,
+    mojo_per_unit: int,
+    coin_record: CoinRecord,
+) -> None:
+    if verbose:
+        print(coin_record)
+    else:
+        chia_amount = Decimal(int(coin_record.coin.amount)) / mojo_per_unit
+        coin_address = encode_puzzle_hash(coin_record.coin.puzzle_hash, address_prefix)
+        confirmed_block = coin_record.confirmed_block_index
+        print(f"Coin 0x{coin_record.name}")
+        print(f"Wallet Address: {coin_address}")
+        print(
+            "Status: "
+            f"{f'Confirmed at block: {confirmed_block}' if coin_record.confirmed_block_index != 0 else 'Pending'}"
+        )
+        print(f"Spent: {f'At Block {coin_record}' if not coin_record.spent else 'No'}")
+        print(f"Amount {'sent'}: {chia_amount} {name}")
+        print(f"Parent Coin ID: 0x{coin_record.coin.parent_coin_info}")
```

Comment:
`parent_coin_info.hex()`

Reply:
i assumed that printing it would auto hex it but ill add that anyway